packages/mermaid/src/diagrams/git/gitGraphTypes.ts

```diff
@@ -0,0 +1,133 @@
+import type { GitGraphDiagramConfig } from '../../config.type.js';
+import type { DiagramDBBase } from '../../diagram-api/types.js';
+
+export interface CommitType {
+  NORMAL: number;
+  REVERSE: number;
+  HIGHLIGHT: number;
+  MERGE: number;
+  CHERRY_PICK: number;
+}
+
+export interface Commit {
+  id: string;
+  message: string;
+  seq: number;
+  type: number;
+  tags: string[];
+  parents: string[];
+  branch: string;
+  customType?: number;
+  customId?: boolean;
+}
+
+export interface GitGraph {
+  statements: Statement[];
+}
+
+export interface Position {
+  x: number;
+  y: number;
+}
+
+export type Statement = CommitAst | BranchAst | MergeAst | CheckoutAst | CherryPickingAst;
+
+export interface CommitAst {
+  $type: 'Commit';
+  id: string;
+  message?: string;
+  tags?: string[];
+  type?: 'NORMAL' | 'REVERSE' | 'HIGHLIGHT';
+}
+
+export interface BranchAst {
+  $type: 'Branch';
+  name: string;
+  order?: number;
+}
+
+export interface MergeAst {
+  $type: 'Merge';
+  branch: string;
+  id?: string;
+  tags?: string[];
+  type?: 'NORMAL' | 'REVERSE' | 'HIGHLIGHT';
+}
+
+export interface CheckoutAst {
+  $type: 'Checkout';
+  branch: string;
+}
+
+export interface CherryPickingAst {
+  $type: 'CherryPicking';
+  id: string;
+  tags?: string[];
+  parent: string;
```

Comment:
nit: Optional args at the end

```suggestion
  parent: string;
  tags?: string[];
```